ncm-ssh/src/main/perl/ssh.pm

```diff
@@ -20,20 +20,22 @@ our $NoActionSupported = 1;
 
 use CAF::Process;
 use CAF::FileEditor;
-use constant SSHD_CONFIG  => "/etc/ssh/sshd_config";
-use constant SSH_CONFIG   => "/etc/ssh/ssh_config";
-use constant SSH_VALIDATE => qw(/usr/sbin/sshd -t -f /proc/self/fd/0);
+use constant DEFAULT_SSHD_CONFIG  => "/etc/ssh/sshd_config";
+use constant DEFAULT_SSHD_PATH    => "/usr/sbin/sshd";
+use constant DEFAULT_SSH_CONFIG   => "/etc/ssh/ssh_config";
 
 # Returns true if $file is a valid SSHD configuration file.
 sub valid_sshd_file
 {
-    my ($self, $file) = @_;
+    my ($self, $file, $cfg) = @_;
+
+    my @cmdline = ( $cfg->{sshd_path} || DEFAULT_SSHD_PATH, '-t', '-f', '/dev/stdin' );
```

Comment:
I used a File::Temp initially, then someone here suggested using /dev/stdin so I updated the commit.

On Red Hat Linux, this is a symlink to /proc/self/fd/0, so it works the same as before. On Solaris there is a symlink to fd/0.

$ ls -l /dev/stdin
lrwxrwxrwx   1 root            6 May 23 05:23 /dev/stdin -> ./fd/0
$ ls -l /dev/fd/0
crw-rw-rw-   1 root     404,   0 Oct 12 16:49 /dev/fd/0
$ echo test | cat /dev/stdin
test